src/snowflake/connector/config_parser/__init__.py

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from pathlib import Path
+from shutil import copyfile
+
+from tomlkit import parse
+
+from ..constants import config_file
+from .parser import ConfigParser
+
+# Create default configuration file, if it doesn't exist
+if not config_file.exists():
```

Comment:
Two questions.

* Should we be doing this check in the module globals?  It means this gets executed when the module is imported which may or may not be the right time.  Also, I'm always wary of module import side effects.  It's sometimes difficult to predict when a module will be imported.  Would it be better to have an explicit call that guarantees the file exists?

* Isn't there a potential race condition here?  What if two processes end up here at the same time?  One find the file doesn't exist but before it can copy, another process enters the same conditional.  [copyfile()](https://docs.python.org/3/library/shutil.html#shutil.copyfile) says the destination will get ovewritten, but doesn't say that the write is atomic, so what's the possibility of file corruption?  Do you think it would be better to expand this and use EAFP principles to ensure atomicity?

Reply:
* Could you propose how you think this should work? When should such a function be called?
* Yes, I forgot about this... We do something similar in [cache.py](https://github.com/snowflakedb/snowflake-connector-python/blob/d6a47498eefee8c125ad37cecc2361f6ef66da45/src/snowflake/connector/cache.py#L523). Essentially make temp file next to target file, write all content and then overwrite target file with temporary one. Would this be enough?

Reply:
+1 to refactor into a function and not run it as module import.

Also, I'd maybe recommend to create the default config file as part of setup.py?

Reply:
> Essentially make temp file next to target file, write all content and then overwrite target file with temporary one. Would this be enough?

Yes, and I think you can do all of that a little more cleanly with higher level `tempfile` and `pathlib` APIs.  I've done this like a. million times 😄 

> +1 to refactor into a function and not run it as module import.

+1, or use an [existing library](https://python-atomicwrites.readthedocs.io/en/latest/).

Reply:
> Also, I'd maybe recommend to create the default config file as part of setup.py?

-1; what happens when we replace setuptools with more modern Python packaging based on declarative `pyproject.toml` files?

Is there a top level entry point in the library that everyone needs to call to initialize things?

Reply:
We synced up offline and decided not to do default config file creation.